src/base/static/components/place-detail/index.js

```diff
@@ -81,95 +72,87 @@ const PlaceDetailContainer = styled("div")(props => ({
-          isSubmitterEditingSupported: this.categoryConfig
-            .submitter_editing_supported,
-        }) ||
-        this.props.hasGroupAbilitiesInDatasets({
-          abilities: ["update"],
-          submissionSet: "places",
-          datasetSlugs: [this.props.collectionId],
-        }),
-      isEditFormSubmitting: false,
-      isSurveyEditFormSubmitting: false,
-    };
-
-    // topOffset = header bar height + padding + geocoding bar height (if enabled).
-    this.topOffset = 80 + (this.props.mapConfig.geocoding_bar_enabled ? 72 : 0);
-  }
-
-  componentDidMount() {
-    // Maintain reset listeners for submission collections in case the detail
-    // view is instantiated before these collections have been fetched.
-    this.props.model.submissionSets[this.supportType].on(
-      "reset",
-      collection => {
-        this.setState({
-          supportModels: serializeBackboneCollection(collection),
-        });
-      },
-    );
-
-    this.props.model.submissionSets[this.surveyType].on("reset", collection => {
-      this.setState({
-        surveyModels: serializeBackboneCollection(collection),
-      });
+  state = {
+    place: fromJS(this.props.getPlace(this.props.placeId)),
+    isSurveyEditFormSubmitting: false,
+    placeRequestType: null,
+  };
+  categoryConfig = getCategoryConfig(
+    this.props.placeConfig,
+    this.state.place.get(constants.LOCATION_TYPE_PROPERTY_NAME),
+  );
+  submitter = this.state.place.get("submitter") || Map();
+  submitterName = this.state.place.get(constants.SUBMITTER)
+    ? this.state.place.get(constants.SUBMITTER).get("name")
+    : this.state.place.get(constants.NAME_PROPERTY_NAME) ||
+      this.props.placeConfig.anonymous_name;
+  isTagBarEditable = this.props.hasGroupAbilitiesInDatasets({
+    abilities: ["update", "destroy", "create"],
+    submissionSet: "tags",
+    datasetSlugs: [this.state.place.get("_datasetSlug")],
+  });
+  isPlaceDetailEditable =
+    this.props.hasUserAbilitiesInPlace({
+      submitter: this.state.place.get(constants.SUBMITTER),
+      isSubmitterEditingSupported: this.categoryConfig
+        .submitter_editing_supported,
+    }) ||
+    this.props.hasGroupAbilitiesInDatasets({
+      abilities: ["update"],
+      submissionSet: "places",
+      datasetSlugs: [this.state.place.get("_datasetSlug")],
     });
+  // topOffset = header bar height + padding + geocoding bar height (if enabled).
+  topOffset = 80 + (this.props.mapConfig.geocoding_bar_enabled ? 72 : 0);
 
+  componentDidMount() {
     this.props.container.scrollTop = 0;
+
     // Fire on mount hook.
     // The on mount hook allows flavors to run arbitrary code after the detail
     // view mounts.
+
+    // TODO
     if (customHooks && customHooks.onDetailViewMount) {
       hooks[customHooks.onDetailViewMount](this.state);
     }
   }
 
+  componentDidUpdate() {
+    const newPlace = fromJS(this.props.getPlace(this.props.placeId));
+
+    if (
```

Comment:
I'm not sure about this giant check here. The idea is that we want to check if any information about a Place has changed (because and editor has edited the place, a support has been added, a comment has been added, etc.) and re-render if so. To do this we check various aspects of the Place's data in the Places duck against the version of the place stored in local state. The checks are complex because we need to look at various properties within in the Place (such as the array of comments) and check if anything has changed.

This check seems smelly, but I also wasn't sure how else to handle the place detail view.